frontend/src/core/cells/cells.ts

```diff
@@ -214,17 +218,39 @@ const {
         : state.cellIds.topLevelIds.indexOf(cellId);
     const insertionIndex = before ? index : index + 1;
 
+    let newCellCode = code;
+
+    if (includeSelectionAsInitialCode) {
+      const id = cellId as CellId;
+      const editorView = state.cellHandles[id].current?.editorView;
+      if (editorView) {
+        const [highlighted, leftover] = extractHighlightedCode(editorView);
+        if (highlighted.length > 0) {
+          newCellCode = highlighted === "" ? code : highlighted;
+          updateEditorCodeFromPython(editorView, leftover);
+          state.cellData = {
+            ...state.cellData,
+            [id]: {
+              ...state.cellData[id],
+              code: leftover,
+              edited: true,
```

Comment:
this might need to be `edited: Boolean(leftover) && state.cellData[id].code !== state.cellData[id].code`,

i noticed a bug with markdown with no selection being made stale, it would be great to prove this in a test: get the test to fail first and then fix this code so the tests passes

Reply:
made a suggestion above that you may not need this.